packages/medusa/src/services/sales-channel.ts

```diff
@@ -170,6 +192,53 @@ class SalesChannelService extends TransactionBaseService<SalesChannelService> {
         })
     })
   }
+
+  /**
+   * Creates a query object to be used for list queries.
+   * @param selector - the selector to create the query from
+   * @param config - the config to use for the query
+   * @return an object containing the query, relations and free-text
+   *   search param.
+   */
+  protected prepareListQuery_(
+    selector: Selector<SalesChannel> & { q?: string },
+    config: FindConfig<SalesChannel>
+  ): {
+    q?: string
+    relations: (keyof SalesChannel)[]
+    query: ExtendedFindConfig<SalesChannel, Selector<SalesChannel>>
+  } {
+    const selector_ = { ...selector }
+    const config_ = { ...config }
+
+    let q: string | undefined
+    if ("q" in selector_) {
+      q = selector_.q
+      delete selector_.q
+    }
+
+    const query = buildQuery<Selector<SalesChannel>, SalesChannel>(
+      selector_,
+      config_
+    )
+
+    if (config_.relations && config_.relations.length > 0) {
+      query.relations = config_.relations
+    }
+
+    if (config_.select && config_.select.length > 0) {
+      query.select = config_.select
+    }
```

Comment:
Is there a specific reason to include this here as well?  The buildquery already includes extraction of `select` and `relations`

Reply:
Yeah true, it was just for the q parameter but i think that i gone a bit to far 🤣 good catch